services/cms-price-check.js

```diff
@@ -0,0 +1,168 @@
+const express = require('express');
+const uuidv4 = require('uuid/v4');
+const priceTable = require('../lib/rxnorm-prices.json');
+
+const router = express.Router();
+
+function getValidCoding(codings, system) {
+  let validCoding = null;
+  codings.forEach((coding) => {
+    if (coding.code && coding.system === system) {
+      validCoding = coding;
+    }
+  });
+  return validCoding;
+}
+
+function getValidCodingFromConcept(medicationCodeableConcept) {
+  let coding = null;
+  if (medicationCodeableConcept && medicationCodeableConcept.coding) {
+    coding = getValidCoding(medicationCodeableConcept.coding, 'http://www.nlm.nih.gov/research/umls/rxnorm');
+  }
+  return coding;
+}
+
+function getValidResource(resource, context) {
+  let coding = null;
+  if (resource.resourceType === 'MedicationOrder' || resource.resourceType === 'MedicationRequest') {
+    // Check if patient in reference from medication resource refers to patient in context
+    if (resource.patient && resource.patient.reference === `Patient/${context.patientId}`) {
+      const { medicationCodeableConcept } = resource;
+      coding = getValidCodingFromConcept(medicationCodeableConcept);
+    }
+  }
+  return coding;
+}
+
+function getValidContextResources(request) {
+  const resources = [];
+  const { context } = request.body;
+  if (context && context.patientId && context.medications) {
+    const medResources = context.medications;
+    medResources.forEach((resource) => {
+      const validResource = getValidResource(resource, context);
+      if (validResource) {
+        resources.push(resource);
```

Comment:
Should we be pushing the `validResource` instead of the resource here ?

Reply:
No, we should push the entire resource so it is processed later when making the CDS Hooks card. I do see how the variable name could make it misleading, so I re-named it for it's real purpose [here](https://github.com/cds-hooks/sandbox-cds-services/pull/7/commits/546ebd819522858ca31dd4912ada441cc7b466ba). Really, `getValidResource` is a chain of methods to check if the `resource` variable is valid enough to process for a possible card. Thanks for catching that!